### ISSUE TITLE
Example and docs for TLS on WAMP/Netty

### ISSUE THREAD

**Issue by mifkamaz** [crossbario/autobahn-java#239]
When using SSL
```I/Websocket: Closed java.security.cert.CertPathValidatorException: Trust anchor for certification path not found.```

URL: wss://heydoc.ru:8443/serv/websocket

If replace (in `WebSocketConnection.WebSocketConnector.run()`)
```
mSocket = SSLSocketFactory.getDefault().createSocket();
```
to
```
SSLContext context = SSLContext.getInstance("TLS");
mSocket = SSLSocketFactory.getDefault().createSocket();
context.init(null, new X509TrustManager[]{new X509TrustManager(){
   public void checkClientTrusted(X509Certificate[] chain, String authType) throws CertificateException {}
   public void checkServerTrusted(X509Certificate[] chain, String authType) throws CertificateException {}
   public X509Certificate[] getAcceptedIssuers() {
                            return new X509Certificate[0];
}}}, new SecureRandom());
mSocket = context.getSocketFactory().createSocket();
```

All ok, but this bad idea (trust all certificates)?

**Comment by om26er** [comment]
Thanks for reporting this bug. This needs investigation, as we haven't fully tested the TLS support, we'll get there and provide proper guide for that.

**Comment by oberstet** [comment]
This isn't a bug, but we should have docs.

See here what to do: https://stackoverflow.com/questions/2642777/trusting-all-certificates-using-httpclient-over-https/6378872#6378872

**Comment by oberstet** [comment]
This is pretty much essential for a production app, definitely for Android, and often also for Netty based backend components.

For Android, being able to use TLS was probably the single most asked for feature (at the raw WebSocket) level.

**Comment by oberstet** [comment]
So actually it would be good to have the demo gallery driver for the WAMP example there also in a variant that is running over TLS.

TLS support is pretty much required for production, so we should have an example/docs

**Comment by om26er** [comment]
Just to clarify, the original bug report was for the old Android implementation. This is a non-issue for the Netty/WAMP implementation. I tried TLS with self-signed certificates using crossbar, wamp works fine. I also tried with the url in this bug report, that too worked fine.

What we don't implement right now is client certification authorization but then we don't support any kind of authorization in ABJ as of yet.
